enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,504 @@
+    or VMs can boot from it. For bare metal, that will generally be a
+    web server visible to the management controllers in the bare metal
+    hosts (because the management controllers are unlikely to be on
+    the same network as the external interfaces for the cluster
+    hosts). For vSphere or other hypervisors, that may be a filesystem
+    on the hypervisor host.
+11. The orchestration tool configures the host to boot the ISO. For
+    bare metal, that means configuring the baseboard management
+    controller (BMC) in each host to expose the ISO through the
+    virtual media interface. For vSphere or other hypervisors, similar
+    settings of the VM need to be adjusted.
+12. The orchestration tool boots the hosts.
+13. The orchestration tool uses `openshift-install wait-for
+    install-complete` to watch the deployment progress and wait for it
+    to complete.
+14. On each host, the startup script in the image selects the correct
+    network settings based on the MAC addresses visible and applies
+    the network settings for each interface.
+15. On node0, the startup script in the image recognizes that it is on
+    node0 and launches the assisted service and the create-cluster
+    service to drive the deployment.
+16. On node0, the create-cluster service waits for the assisted
+    service to start, then creates an InfraEnv using the API.
+17. On node0, the create-cluster service waits for all of the known
+    agents to be registered with the service.
+18. On all nodes (including node0), the startup script in the image
+    connects to the assisted service on node0 to find the InfraEnv ID
+    for the cluster being deployed (there will only be 1). After it
+    has the ID, the startup script launches the assisted agent with
+    that ID and configured to connect to the assisted service on
+    node0.
+19. All of the agents register their hosts with the assisted service.
+20. The assisted service performs its standard validations (network
+    connectivity, storage space, RAM, etc.) for all of the hosts.
+21. On node0, the create-cluster service sees that all known agents
+    have registered and the hosts have passed validation. It uses the
+    assisted service API to trigger the deployment to proceed.
+22. The assisted service performs the cluster deployment in its usual
+    way, except that the nodes will use the ISOs they booted from to
+    install RHCOS instead of fetching a new image from the assisted
+    service.
+23. The `openshift-install wait-for install-complete` process sees the
+    OpenShift API become available and starts using it to watch the
+    cluster complete deployment, combining the information provided by
+    `ClusterVersion` and `ClusterOperator` resources with the assisted
+    service progress API.
+24. On node0, when all of the other hosts have been deployed and
+    bootstrapping is complete, the agent reboots node0.
+25. The `openshift-install wait-for install-complete` process loses
```

Comment:
we need new sub command - wait-for-agent-bootstrap? :) or something similar?

Reply:
also for showing events (what the progress will show? Events or cluster status?)
Also, another required sub command will be to download logs

Reply:
I think this is covered in the section describing the `agent wait-for` command, currently on lines 203-217.